tmail-backend/event-bus-redis/src/main/scala/org/apache/james/events/RedisEventBusClientFactory.scala

```diff
@@ -6,16 +6,19 @@ import io.lettuce.core.pubsub.api.reactive.RedisPubSubReactiveCommands
 import io.lettuce.core.{AbstractRedisClient, RedisClient}
 import jakarta.annotation.PreDestroy
 import jakarta.inject.{Inject, Singleton}
-import org.apache.james.backends.redis.{ClusterRedisConfiguration, MasterReplicaRedisConfiguration, RedisConfiguration, StandaloneRedisConfiguration}
-
-import scala.jdk.CollectionConverters._
+import org.apache.james.backends.redis.{ClusterRedisConfiguration, MasterReplicaRedisConfiguration, RedisClientFactory, RedisConfiguration, SentinelRedisConfiguration, StandaloneRedisConfiguration}
 
 class RedisEventBusClientFactory @Singleton() @Inject()
-(redisConfiguration: RedisConfiguration) {
+(redisConfiguration: RedisConfiguration, redisClientFactory: RedisClientFactory) {
   val rawRedisClient: AbstractRedisClient = redisConfiguration match {
-    case standaloneConfiguration: StandaloneRedisConfiguration => RedisClient.create(standaloneConfiguration.redisURI)
-    case masterReplicaRedisConfiguration: MasterReplicaRedisConfiguration => RedisClient.create(masterReplicaRedisConfiguration.redisURI.value.last)
-    case clusterRedisConfiguration: ClusterRedisConfiguration => RedisClusterClient.create(clusterRedisConfiguration.redisURI.value.asJava)
+    case standaloneConfiguration: StandaloneRedisConfiguration => redisClientFactory.createStandaloneClient(standaloneConfiguration)
```

Comment:
This block code is similar to `RedisRateLimiter` class ?
Should we extract it to util method ?
`getRedisClientFromConfiguration`

Reply:
I don't see any similarity here